test/services/payout-service-test.ts

```diff
@@ -0,0 +1,261 @@
+            testData.issuer.address,
+            testData.payoutManager.address,
+            [
+                testData.assetFactory.address,
+                testData.assetTransferableFactory.address,
+                testData.assetSimpleFactory.address
+            ]
+        );
+
+        expect(payoutsAfterCreation.length).to.be.equal(3);
+        
+        const assetBasicPayout: PayoutForIssuerItem = 
+            payoutsAfterCreation.filter(item => item.asset == assetBasic.address)[0];
+        const assetTransferablePayout: PayoutForIssuerItem = 
+            payoutsAfterCreation.filter(item => item.asset == assetTransferable.address)[0];
+        const assetSimplePayout: PayoutForIssuerItem = 
+            payoutsAfterCreation.filter(item => item.asset == assetSimple.address)[0];
+        
+        assertPayoutForIssuerResponse(
+            expectedAssetSimplePayoutId,
+            assetSimple.address,
+            issuerOwnerAddress,
+            payoutInfo,
+            assetSimplePayout
+        );
+        assertPayoutForIssuerResponse(
+            expectedAssetTransferablePayoutId,
+            assetTransferable.address,
+            issuerOwnerAddress,
+            payoutInfo,
+            assetTransferablePayout
+        );
+        assertPayoutForIssuerResponse(
+            expectedAssetBasicPayoutId,
+            assetBasic.address,
+            issuerOwnerAddress,
+            payoutInfo,
+            assetBasicPayout
+        );
+
+        const aliceAddress = await testData.alice.getAddress();
+        const payoutStatesForInvestor = await testData.payoutService.getPayoutStatesForInvestor(
+            aliceAddress,
+            testData.payoutManager.address,
+            [ BigNumber.from(0), BigNumber.from(1), BigNumber.from(2) ]
+        );
+        expect(payoutStatesForInvestor.length).to.be.equal(3);
+        expect(payoutStatesForInvestor[0].payoutId).to.be.equal(0);
+        expect(payoutStatesForInvestor[0].investor).to.be.equal(aliceAddress);
+        expect(payoutStatesForInvestor[0].amountClaimed).to.be.equal(0);
+        expect(payoutStatesForInvestor[1].payoutId).to.be.equal(1);
+        expect(payoutStatesForInvestor[1].investor).to.be.equal(aliceAddress);
+        expect(payoutStatesForInvestor[1].amountClaimed).to.be.equal(0);
+        expect(payoutStatesForInvestor[2].payoutId).to.be.equal(2);
+        expect(payoutStatesForInvestor[2].investor).to.be.equal(aliceAddress);
+        expect(payoutStatesForInvestor[2].amountClaimed).to.be.equal(0);
+    });
+
+    /* HELPERS */
+
+    interface PayoutInfo {
+        totalAssetAmount: BigNumber,
+        ignoredAssetAddresses: string[],
+        payoutInfo: string,
+        assetSnapshotMerkleRoot: string,
+        assetSnapshotMerkleDepth: BigNumber,
+        assetSnapshotBlockNumber: BigNumber,
+        assetSnapshotMerkleIpfsHash: string,
+        totalRewardAmount: BigNumber,
+        rewardAsset: string
+    };
+
+    interface PayoutForIssuerItem {
```

Comment:
Is it possible to use generated interface from `Payout` defined in `Structs` contract?

Reply:
seems it's not possible to do this , did a bit of googling.
For the rest of the comments I've fixed them

Reply:
Yeah, it's very annoying - typechain generated values return something that is both array and an object, e.g. `[string, number] & { something: string, other: number }`. (Java|Type)Script black magic I guess 🤷‍♂️